packages/components/addon/components/hds/table/index.js

```diff
@@ -179,4 +182,57 @@ export default class HdsTableIndexComponent extends Component {
       onSort(this.sortBy, this.sortOrder);
     }
   }
+
+  @action
+  addCheckbox(checkbox, scope) {
+    if (scope === 'row') {
+      this.checkboxRowItems.push(checkbox);
```

Comment:
This works for the first rendering of the table, but what happens if rows are added/removed: then this array goes out of sync with what's in the DOM (same problem that happened with the `Tabs` and `didInsert`. 
One thing we discussed was to have the callback check the status of the rows **when** the user triggered the callback: in this way we always had the exact status in the DOM, not the one in memory that may not correspond to the one in the DOM.

(I've pushed a commit with a couple of examples for filtered and paginated data, you can play with it and you'll see what I mean)

Reply:
Pushed up changes to now use the DOM vs tracking in memory.

The examples you added still don't work because when the DOM changes, the status of the header doesn't get updated. I need to see how to trigger the `setTableStatus` function next.

Reply:
Do you have any suggestions on the best way to trigger it? (I'm still experimenting to see if I can figure it out.)

Reply:
The only things I can think of are
- a dedicated function that the consumers can invoke to "refresh" the status of the "select all" whenever the rows are updated 
- re-introduce back the ember modifiers for `did-insert` + `will-destroy` and keep track ourselves of changes to the DOM structure (this may lead to resolve the need for some DOM traversing that we're doing in the code); but in theory these modifiers are deprecated in "modern" Ember, but frankly I don't see them go anywhere in the near future 

I have set up a sync meeting for Friday where we can brainstorm a bit about this a bit more.

/cc @alex-ju

Reply:
Thanks.

I can't figure out how the first one would work to expose a function.

I've been trying the 2nd one to add back `did-insert` & `will-destroy` but so far there's an issue passing the callback function from the table to the checkbox I've been trying to figure out.

Reply:
I pushed up updates to use `did-insert` & `will-destroy`. They are just calling the `setSelectAllState` function right now. It appears to fix the problems with the inline filter and pagination examples in terms of the header checkbox state but the state of the checkboxes in the `tbody` is still lost when the DOM updates.

Reply:
> I pushed up updates to use `did-insert` & `will-destroy`. They are just calling the `setSelectAllState` function right now. It appears to fix the problems with the inline filter and pagination examples in terms of the header checkbox state 

Great, it means we're on the right track :)

> but the state of the checkboxes in the `tbody` is still lost when the DOM updates.

Which is expected, no? Seems the ambassadors agreed on the fact that preserving the status is a consumer's concern.

One thing I noticed: in this example the "select all" doesn't work anymore
<img width="841" alt="modified" src="https://github.com/hashicorp/design-system/assets/686239/66f7ece1-f668-44f5-8879-77d3b1d02290">

Reply:
Some other pointers:
- https://github.com/hashicorp/cloud-ui/pull/7235 (shared by @meirish)
- https://www.material-react-table.com/docs/guides/row-selection (see in the devtools the console.log response on selection)
- https://react-table-library.com/?path=/docs/compact-table--select (see console in devtools)
- https://www.ag-grid.com/react-data-grid/row-selection/#selection-events
- https://tanstack.com/table/v8/docs/api/features/row-selection#onrowselectionchange
  - https://codesandbox.io/p/devbox/tanstack-table-example-row-selection-2llvty

Reply:
Btw, select all should now be working again. I had a mistake in one of the function names passed.

Reply:
I could also probably write a function to make the selections persist now also if you think it would be helpful.

Reply:
Closing this conversation now